wrapanapi/msazure.py

```diff
@@ -401,12 +423,14 @@ def list_load_balancer(self):
     def does_load_balancer_exist(self, lb_name):
         return lb_name in self.list_load_balancer()
 
-    def remove_diags_container(self):
-        for container in self.container_client.list_containers():
+    def remove_diags_container(self, container_client=None):
+        container_client = container_client or self.container_client
```

Comment:
I'll keep this code using it's own container_client as we need to loop though all containers and leave the default as is